go-controller/pkg/ovn/pods.go

```diff
@@ -149,64 +183,99 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 	}
 
 	logrus.Infof("Deleting pod: %s", pod.Name)
-	logicalPort := fmt.Sprintf("%s_%s", pod.Namespace, pod.Name)
-	out, stderr, err := util.RunOVNNbctlHA("--if-exists", "lsp-del",
-		logicalPort)
-	if err != nil {
-		logrus.Errorf("Error in deleting pod logical port "+
-			"stdout: %q, stderr: %q, (%v)",
-			out, stderr, err)
+
+	podLogicalPorts := make(map[string]string)
+	switches := oc.getNetworkNamesFromPodAnnotations(pod.Annotations)
+	if switches != nil {
+		for _, logicalSwitch := range switches {
+			podLogicalPorts[logicalSwitch] = fmt.Sprintf("%s_%s_%s", pod.Namespace, pod.Name, logicalSwitch)
+		}
 	}
+	podLogicalPorts[""] = fmt.Sprintf("%s_%s", pod.Namespace, pod.Name)
 
-	ipAddress := oc.getIPFromOvnAnnotation(pod.Annotations["ovn"])
+	for logicalSwitch, logicalPort := range podLogicalPorts {
+		out, stderr, err := util.RunOVNNbctlHA("--if-exists", "lsp-del",
+			logicalPort)
+		if err != nil {
+			logrus.Errorf("Error in deleting pod logical port "+
+				"stdout: %q, stderr: %q, (%v)",
+				out, stderr, err)
+		}
 
-	delete(oc.logicalPortCache, logicalPort)
+		var ipAddress string
+		if logicalSwitch == "" {
+			ipAddress = oc.getIPFromOvnAnnotation(pod.Annotations["ovn"])
+		} else {
+			networkAnntationMap := oc.getNetworkInfoFromOvnAnnotation(pod.Annotations["ovn_extra"], logicalSwitch)
+			ipAddress = oc.getIPFromOvnAnnotationMap(networkAnntationMap)
+		}
 
-	oc.lspMutex.Lock()
-	delete(oc.lspIngressDenyCache, logicalPort)
-	delete(oc.lspEgressDenyCache, logicalPort)
-	delete(oc.logicalPortUUIDCache, logicalPort)
-	oc.lspMutex.Unlock()
+		delete(oc.logicalPortCache, logicalPort)
 
-	if !oc.portGroupSupport {
-		oc.deleteACLDenyOld(pod.Namespace, pod.Spec.NodeName, logicalPort,
-			"Ingress")
-		oc.deleteACLDenyOld(pod.Namespace, pod.Spec.NodeName, logicalPort,
-			"Egress")
+		oc.lspMutex.Lock()
+		delete(oc.lspIngressDenyCache, logicalPort)
+		delete(oc.lspEgressDenyCache, logicalPort)
+		delete(oc.logicalPortUUIDCache, logicalPort)
+		oc.lspMutex.Unlock()
+
+		if !oc.portGroupSupport {
+			oc.deleteACLDenyOld(pod.Namespace, pod.Spec.NodeName, logicalPort,
+				"Ingress")
+			oc.deleteACLDenyOld(pod.Namespace, pod.Spec.NodeName, logicalPort,
+				"Egress")
+		}
+		oc.deletePodFromNamespaceAddressSet(pod.Namespace, ipAddress)
 	}
-	oc.deletePodFromNamespaceAddressSet(pod.Namespace, ipAddress)
+
 	return
 }
 
-func (oc *Controller) addLogicalPort(pod *kapi.Pod) {
+func (oc *Controller) addLogicalPort(pod *kapi.Pod, logicalSwitch string) string {
```

Comment:
IMO, better to use newAnotation as a named return value

Reply:
done